integration-test/src/test/java/org/jboss/pnc/integration/ProductReleaseRestTest.java

```diff
@@ -0,0 +1,176 @@
+package org.jboss.pnc.integration;
+
+import com.jayway.restassured.http.ContentType;
+import com.jayway.restassured.response.Response;
+import com.jayway.restassured.response.ValidatableResponse;
+
+import org.assertj.core.api.Assertions;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.junit.InSequence;
+import org.jboss.pnc.auth.AuthenticationProvider;
+import org.jboss.pnc.auth.ExternalAuthentication;
+import org.jboss.pnc.common.util.IoUtils;
+import org.jboss.pnc.integration.matchers.JsonMatcher;
+import org.jboss.pnc.integration.template.JsonTemplateBuilder;
+import org.jboss.pnc.integration.deployments.Deployments;
+import org.jboss.pnc.test.category.ContainerTest;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+
+import static com.jayway.restassured.RestAssured.given;
+import static org.jboss.pnc.integration.env.IntegrationTestEnv.getHttpPort;
+
+@RunWith(Arquillian.class)
+@Category(ContainerTest.class)
+public class ProductReleaseRestTest {
+
+    public static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private static final String PRODUCT_REST_ENDPOINT = "/pnc-rest/rest/product/";
+    private static final String PRODUCT_VERSION_REST_ENDPOINT = "/pnc-rest/rest/product/%d/version/";
+    private static final String PRODUCT_MILESTONE_REST_ENDPOINT = "/pnc-rest/rest/product-milestone/";
+    private static final String PRODUCT_RELEASE_REST_ENDPOINT = "/pnc-rest/rest/product-release/";
+    private static final String PRODUCT_RELEASE_SPECIFIC_REST_ENDPOINT = PRODUCT_RELEASE_REST_ENDPOINT + "%d";
+
+    private static int productId;
+    private static int productVersionId;
+    private static int productMilestoneId;
+    private static int productReleaseId;
+    private static int newProductReleaseId;
+
+    private static AuthenticationProvider authProvider;
+
+
+    @Deployment(testable = false)
+    public static EnterpriseArchive deploy() {
+        EnterpriseArchive enterpriseArchive = Deployments.baseEar();
+        logger.info(enterpriseArchive.toString(true));
+        return enterpriseArchive;
+    }
+
+    @Before
+    public void prepareData() {
+        try {
+            InputStream is = this.getClass().getResourceAsStream("/keycloak.json");
+            ExternalAuthentication ea = new ExternalAuthentication(is);
```

Comment:
@BeforeClass happens before Arquillian deployment. Take a look at other 
tests with AtomicBoolean trick or maybe you could come up with some 
better idea.

On 04/10/2015 03:36 PM, Paul Gier wrote:
>
> In 
> integration-test/src/test/java/org/jboss/pnc/integration/ProductReleaseRestTest.java 
> <https://github.com/project-ncl/pnc/pull/282#discussion_r28144376>:
>
> > +
> > +    private static AuthenticationProvider authProvider;
> > +
> > +
> > +    @Deployment(testable = false)
> > +    public static EnterpriseArchive deploy() {
> > +        EnterpriseArchive enterpriseArchive = Deployments.baseEar();
> > +        logger.info(enterpriseArchive.toString(true));
> > +        return enterpriseArchive;
> > +    }
> > +
> > +    @Before
> > +    public void prepareData() {
> > +        try {
> > +            InputStream is = this.getClass().getResourceAsStream("/keycloak.json");
> > +            ExternalAuthentication ea = new ExternalAuthentication(is);
>
> You're right, I had just copied it from the existing product versions 
> rest test. Looks like the auth setup is kind of inconsistent between 
> out test classes currently. I'll can change it to "@BeforeClass" and 
> see if that works.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/project-ncl/pnc/pull/282/files#r28144376>.
>